dotnet/src/SemanticKernel.Core/Planning/InstrumentedPlan.cs

```diff
@@ -34,16 +34,29 @@ internal sealed class InstrumentedPlan : IPlan
     /// <summary>
     /// Initialize a new instance of the <see cref="InstrumentedPlan"/> class.
     /// </summary>
-    /// <param name="plan">Instance of <see cref="IPlan"/> to decorate.</param>
+    /// <param name="plan">Instance of <see cref="Plan"/> to decorate.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     public InstrumentedPlan(
-        IPlan plan,
+        Plan plan,
         ILoggerFactory? loggerFactory = null)
     {
         this._plan = plan;
         this._logger = loggerFactory is not null ? loggerFactory.CreateLogger(typeof(InstrumentedPlan)) : NullLogger.Instance;
     }
 
+    /// <summary>
+    /// Initialize a new instance of the <see cref="InstrumentedPlan"/> class.
+    /// </summary>
+    /// <param name="plan">Instance of <see cref="IPlan"/> to decorate.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    [Obsolete("Use the other concrete class Plan constructor instead")]
```

Comment:
This class is already internal, it's possible to initialize it only on our side. In this case, `Obsolete` flag is not required on constructor, and we can remove this constructor completely.

Reply:
I need to obsolete this one as this method need to be removed in the future, important to keep track of it as well due to the Warning it gives when I remove it.

![image](https://github.com/microsoft/semantic-kernel/assets/19890735/38c4a687-1573-460f-adff-48556534998c)